src/Aspire.Hosting/Dcp/DcpExecutor.cs

```diff
@@ -1037,9 +1037,11 @@ private void PrepareProjectExecutables()
                     }
 
                     projectLaunchConfiguration.DisableLaunchProfile = project.TryGetLastAnnotation<ExcludeLaunchProfileAnnotation>(out _);
-                    if (!projectLaunchConfiguration.DisableLaunchProfile && project.TryGetLastAnnotation<LaunchProfileAnnotation>(out var lpa))
+
+                    // Use the effective launch profile which has fallback logic
+                    if (!projectLaunchConfiguration.DisableLaunchProfile && project.GetEffectiveLaunchProfile() is NamedLaunchProfile namedLaunchProfile)
```

Comment:
This is the meat of the change.

Reply:
This change seems overall better since we look for the `LaunchProfileAnnotation` and other places (`DefaultLaunchProfileAnnotation` and the launch profile itself) and `LaunchProfileAnnotation` will still take precendence.

The only weridness is that we now may be respecting launch profiles when we weren't before (e.g. things that didn't set a `LaunchProfileAnnotation`). That seems like a bug in that case anyway, so maybe it's fine?

It seems like Functions projects were the only category that hit this issue anyway?